common/config/rush/pnpm-lock.yaml

```diff
@@ -465,7 +663,7 @@ packages:
       integrity: sha512-CxaMaEjwtsmIhWtjHyGimKO7RmES0YxPqGQ9+jKqGygNlhG5NYHktDaiQu6w7k3g+I51VaLXtVSt+BVFd6VWfQ==
   /@azure/identity/1.2.5_debug@4.3.2:
     dependencies:
-      '@azure/core-http': 1.2.6
+      '@azure/core-http': 1.2.3
```

Comment:
weird to see a downgraded package

Reply:
I think the issue with the package downgrades is our `pnpm-lock.yaml` contains multiple versions of the same package which match the requested version spec, and this seems to trigger non-deterministic behavior in rush+pnpm.

For `@azure/core-http`, our `pnpm-lock.yaml` contains both `1.2.6` and `1.2.3`.  `1.2.6` is the latest, but `azure-iothub` pins to `1.2.3`:

https://github.com/Azure/azure-iot-sdk-node/blob/fc6871228c975c241e9eb02d92a0af95d32cf9dd/service/package.json#L11

If we changed `azure-iothub` to depend on `^1.2.3` I believe this problem would go away.

Reply:
Also, it appears `azure-iothub` is only used in a single sample, so maybe the package could be removed entirely from the rush package graph?

https://github.com/Azure/azure-sdk-for-js/search?q=azure-iothub

Reply:
Looking at this further, I think the new behavior may actually be more correct.  `identity/1.2.5_debug@4.3.2` is only used by `azure-iothub@1.14.3`, which pins `@azure/core-http` to `1.2.3`.  So it seems more correct for `identity/1.2.5_debug@4.3.2` to depend on `@azure/core-http@1.2.3` rather than `@azure/core-http@1.2.6`.  I don't know whether this is an intentional improvement in rush/pnpm, or if it's just unspecified behavior that happened to flip to the "more correct" version this run.